unittests/BuildSystem/BuildExecutionQueueTest.cpp

```diff
@@ -0,0 +1,152 @@
+
+#include "gtest/gtest.h"
+
+using namespace llbuild;
+using namespace llbuild::core;
+using namespace llbuild::buildsystem;
+
+namespace {
+class PhonyCommand : public ExternalCommand {
+  public:
+    using ExternalCommand::ExternalCommand;
+
+    virtual bool shouldShowStatus() override { return false; }
+
+    virtual void getShortDescription(SmallVectorImpl<char> &result) override {
+      llvm::raw_svector_ostream(result) << getName();
+    }
+
+    virtual void getVerboseDescription(SmallVectorImpl<char> &result) override {
+      llvm::raw_svector_ostream(result) << getName();
+    }
+
+    virtual bool executeExternalCommand(BuildSystemCommandInterface& bsci,
+        Task* task,
+        QueueJobContext* context) override {
+      return true;
+    }
+};
+  
+class SimpleBuildExecutionQueueDelegate : public BuildExecutionQueueDelegate {
+
+  public:
+    SimpleBuildExecutionQueueDelegate() { 
+    }
+
+    virtual void commandJobStarted(Command* command) override {
+    }
+
+    virtual void commandJobFinished(Command* command) override {
+    }
+
+    virtual void commandProcessStarted(Command* command,
+        ProcessHandle handle) override {
+    }
+
+    virtual void commandProcessHadError(Command* command, ProcessHandle handle,
+        const Twine& message) override {
+    }
+
+    virtual void commandProcessFinished(Command* command, ProcessHandle handle,
+        int exitStatus) override {
+    }
+};
+
+struct LaneBasedExecutionQueueJobContext {
+    QueueJob& job;
+};
+
+class BuildExecutionQueueTest : public testing::Test {
+  protected:
+    SimpleBuildExecutionQueueDelegate delegate;
+    std::shared_ptr<BuildExecutionQueue> queue;
+    std::unique_ptr<PhonyCommand> phony;
+    std::string stdoutString;
+    std::string stderrString;
+    std::string expectedStdout;
+    std::string expectedStderr;
+
+    BuildExecutionQueueTest() {
+      queue = std::shared_ptr<BuildExecutionQueue>(createLaneBasedExecutionQueue(delegate, 1));
+      phony = llvm::make_unique<PhonyCommand>("Phony");
+    }
+
+    virtual ~BuildExecutionQueueTest() {
+      EXPECT_EQ(stderrString, expectedStderr);
+      EXPECT_EQ(stdoutString, expectedStdout);
+    }
+
+    void executeProcessTester(std::string exe, StringRef output, StringRef error) {
+      expectedStdout = output;
+      expectedStderr = error;
+      // HACK.
+      auto job = QueueJob { phony.get(), [](auto context){} };
+      LaneBasedExecutionQueueJobContext context{ job }; 
+      queue->executeProcess(reinterpret_cast<QueueJobContext*>(&context), {exe}, {}, 
+          [&](StringRef output){ stdoutString.append(output); },
+          [&](StringRef error){ stderrString.append(error); });
+    }
+};
+
+std::string fixturePath(StringRef name) {
+  SmallString<16> basePath = llvm::sys::path::parent_path(__FILE__);
+  llvm::sys::path::append(basePath, "/Inputs/BuildExecutionQueueExecuteProcessInputs/");
+  llvm::sys::path::append(basePath, name); 
+  return basePath.str();
+}
```

Comment:
@ddunbar this is not working when executed via `lit` or `ninja test` but works when binary is executed directly. Any other way to get the path?